### PR TITLE
Update Panel

### PR DESCRIPTION
**Pull request by jonnystening**
Updates styling to use new tokens

Uses Context, removes pop-drilling

Removes sub-component props and PropType checking

Exports sub-components without parent component prefix (e.g. PanelHeading --> Heading)

### CODE REVIEW

File: components/panel/src/PanelBody.js [hunk 1]
```diff
@@ -2,23 +2,16 @@
...
 // Component
 // ==============================
 
-export const PanelBody = ({ appearance, ...props }) => {
-	const { breakpoints, panel } = useTheme();
+export const PanelBody = props => {
+	const {
+		LAYOUT: { breakpoints },
+	} = useTheme();
 	const mq = paint(breakpoints);
```
Comment (jossmac) [inline components/panel/src/PanelBody.js]: Not directly relate to this PR, but it seems odd that we're passing `paint` around and creating the helper utility inside each component. Might be a good time to abstract it e.g.

```jsx
// Somewhere in Core?
export const useMediaQuery = () => {
  const { LAYOUT } = useTheme();

  return paint(LAYOUT.breakpoints);
}

// Use
export const Well = props => {
  const mq = useMediaQuery();

  return (
    <div css={mq({
      query: 'reliant',
      styles: ['can', 'go', 'here']
    })} {...props} />
  );
}
```

Comment (jaortiz) [reply components/panel/src/PanelBody.js]: implemented in separate pr :) 
https://github.com/WestpacGEL/GEL/pull/141

File: components/panel/src/PanelBody.js [hunk 2]
```diff
@@ -1,24 +1,14 @@
...
 
 // ==============================
 // Component
 // ==============================
 
-export const PanelBody = ({ appearance, ...props }) => {
-	const { breakpoints, panel } = useTheme();
-	const mq = paint(breakpoints);
+export const PanelBody = props => {
+	const mq = useMediaQuery(breakpoints);
```
Comment (simonswiss) [inline components/panel/src/PanelBody.js]: We're not passing `breakpoints` anymore here, they're provided in the `useMediaQuery` hook.

```suggestion
      const mq = useMediaQuery();
```

Comment (jonnystening) [reply components/panel/src/PanelBody.js]: My mistake. Was late 😬

**Pull request merged**